### PR TITLE
STRING_AGG with cast

### PR DESCRIPTION
**Pull request by devzhr**

### CODE REVIEW

File: htdocs/core/db/pgsql.class.php [hunk 1]
```diff
@@ -169,7 +170,8 @@ public static function convertSQLFromMysql($line, $type = 'auto', $unescapeslash
 			// Replace group_concat(x) or group_concat(x SEPARATOR ',') with string_agg(x, ',')
 		    $line = preg_replace('/GROUP_CONCAT/i', 'STRING_AGG', $line);
 			$line = preg_replace('/ SEPARATOR/i', ',', $line);
-			$line = preg_replace('/STRING_AGG\(([^,\)]+)\)/i', 'STRING_AGG(\\1, \',\')', $line);
+			$line = preg_replace('/STRING_AGG\(([^,\)]+)/i', 'STRING_AGG(CAST(\1 AS VARCHAR(8000))', $line);
+			$line = preg_replace('/"@"/', '\'@\'', $line);
```
Comment (eldy) [inline htdocs/core/db/pgsql.class.php]: Do you have example of request that need this ?

Comment (zouhair2015) [reply htdocs/core/db/pgsql.class.php]: for example this module "dolibarr_module_supplierorderfromorder" it does not work with pgsql and the problem on dolibarr it is not on the module

Comment (eldy) [reply htdocs/core/db/pgsql.class.php]: This looks more than the module does not write correctly its requests. May be our rules to write correctly SQL request are not complete: https://wiki.dolibarr.org/index.php/Language_and_development_rules#SQL_Coding_rules

Can you provide the exact sql request this module writes ?

Comment (zouhair2015) [reply htdocs/core/db/pgsql.class.php]: SELECT p.rowid, p.ref, p.label, cd.description, p.price, SUM(cd.qty) as qty, cd.buy_price_ht, p.price_ttc, p.price_base_type,p.fk_product_type, p.tms as datem, p.duration, p.tobuy, p.seuil_stock_alerte, p.finished, cd.rang, GROUP_CONCAT(cd.rowid SEPARATOR "@") as lineid, ( SELECT SUM(s.reel) FROM llx_product_stock s WHERE s.fk_product=p.rowid ) as stock_physique, p.desiredstock FROM llx_product as p LEFT OUTER JOIN llx_commandedet as cd ON (p.rowid = cd.fk_product) WHERE p.fk_product_type IN (0,1) AND p.entity IN (1) AND cd.fk_commande = 18 AND p.tobuy = 1 GROUP BY p.rowid, p.ref, p.label, p.price, p.price_ttc, p.price_base_type,p.fk_product_type, p.tms, p.duration, p.tobuy, p.seuil_stock_alerte, cd.rang, cd.description ORDER BY cd.rang ASC LIMIT 1001

Comment (eldy) [reply htdocs/core/db/pgsql.class.php]: Here the bug seems to be in the external module. The module generate a sql with 
 GROUP_CONCAT(cd.rowid SEPARATOR "@") 
so he ask to concat field rowid that is int with string "@" taht is string. 
This is not a good SQL request. It is the responsability of the module to fix its SQL request by adding a CAST around the cd.rowid:   CAST(cd.rowid as CHAR(64))

**Pull request closed without merging**